src/devices/esp8266.c

```diff
@@ -0,0 +1,848 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "at.h"
+#include "esp8266.h"
+#include "macros.h"
+#include "mem_mang.h"
+#include "printk.h"
+#include "serial_buffer.h"
+#include "taskUtil.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define _AT_CMD_DELIM		"\r\n"
+#define _AT_DEFAULT_QP_MS	250
+#define _AT_QP_PRE_INIT_MS     	500
+#define _AT_QP_STANDARD_MS	1	/* Can probably be 0 */
+#define _AUTOBAUD_TRIES		3
+#define _TIMEOUT_LONG_MS	5000
+#define _TIMEOUT_MEDIUM_MS	500
+#define _TIMEOUT_SHORT_MS	50
+#define _TIMEOUT_SUPER_MS	30000
+
+static const enum log_level serial_dbg_lvl = INFO;
+
+/* STIEG: Temp until we write *_create methods for serial_buff and at_info */
+struct serial_buffer _serial_cmd_buff;
+struct at_info _ati;
+
+/**
+ * Internal state of our driver.
+ */
+static struct {
+        struct serial_buffer *scb;
+        struct at_info *ati;
+        void (*init_cb)(enum dev_init_state); /* STIEG: Put in own struct? */
+        enum dev_init_state init_state;
+} state;
+
+static void cmd_failure(const char *cmd_name, const char *msg)
+{
+        pr_warning("[esp8266] ");
+        pr_warning(cmd_name);
+        if (msg) {
+                pr_info_str_msg(" failed with msg: ", msg);
+        } else {
+                pr_warning(" failed.\r\n");
+        }
+}
+
+/**
+ * Sets up the internal state of the driver.  Must be called before init.
+ */
+static bool _setup(struct Serial *s, const size_t max_cmd_len)
+{
+        /* Check if already initialized.  If so, do nothing */
+        if (state.ati)
```

Comment:
Do we have anything firm about naming of module level static variables? When I saw this, I immediately searched 
for the declaration of a local variable.